user-operator/src/main/java/io/strimzi/operator/user/operator/KafkaUserQuotasOperator.java

```diff
@@ -206,7 +214,13 @@ public void delete(String username) {
 
         if (data != null)   {
             log.debug("Deleting quotas for user {}", username);
-            zkClient.writeData("/config/users/" + username, removeQuotasFromJsonUser(data));
+            byte[] deleteJson = removeQuotasFromJsonUser(data);
+            if (jsonConfigEmpty(deleteJson)) {
```

Comment:
So remove `removeQuotasFromJsonUser` takes a `byte[]`, parses it as JSON, removes something, serializes it and then `jsonConfigEmpty()` parses it again? That's not ideal. Is there a reason why `removeQuotasFromJsonUser()` doesn't return the `JsonObject` and let the caller serialize it if&when they need to? `jsonConfigEmpty` would then take a `JsonObject`, but that's better than passing around these `byte[]` and having each method parse and reparse it.